packages/block-editor/src/components/element-button/index.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+const getElementProps = ( props, ref ) => {
+	const newProps = { ...props };
+	newProps.className += ' wp-element-button';
+	newProps.ref = ref;
+	return newProps;
+};
+
+const ElementButton = forwardRef( ( props, ref ) => {
```

Comment:
Would it be possible to introduce a single factory (maybe a hook) that returns the component? Something like

```js
createStylesElement( 'button' ); // for the regular button
createStylesElement( 'button', 'a' ); // for the button as link
```

Reply:
One public method to document and maintain for all elements.

Reply:
It’s very similar to `useBlockProps` for the block wrapper in `edit`. If I recall correctly it also started as a list of predefined components to use with blocks. This is what made me think about more general approach. Anyway, like I said, I don’t think I know enough about the problem so you I might be wrong.

Reply:
Yeah I like that idea, let me give it a go. Thanks for the feedback :)

Reply:
I like the idea of starting with just a class (without an explicit API yet), but I'm personally unsure about the `use` pattern here. The `use` pattern is more difficult to use for third-party developers, it requires knowledge about the about (classname prop) and potentially knowing how to merge things.

Reply:
That’s a good point. Don’t know what to make of it. :)

Reply:
I think we can just massively simplify this and just add a const for a class name to be used across components. I'll see if we can find a way to share this between client and server too...

Reply:
That’s even better 👍